src/apps/api/views/competitions.py

```diff
@@ -211,59 +210,107 @@ def email_all_participants(self, request, pk):
         batch_send_email.apply_async((comp.pk, content))
         return Response({}, status=status.HTTP_200_OK)
 
-    @action(detail=True, methods=['GET'])
-    def get_csv(self, request, pk):
+    def collect_leaderboard_check_permissions(self, request, competition):
```

Comment:
this permissions check already exists on `Competition` model I believe. let's make try to move functions to models